File: src/credentials/sso/ssoCache.ts
/*!
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { SsoClientRegistration } from './ssoClientRegistration'
import { SsoAccessToken } from './ssoAccessToken'

export interface SsoCache {
    loadClientRegistration(ssoRegion: string): SsoClientRegistration | undefined
    saveClientRegistration(ssoRegion: string, registration: SsoClientRegistration): void
    invalidateClientRegistration(ssoRegion: string): void

    loadAccessToken(ssoUrl: string): SsoAccessToken | undefined
    saveAccessToken(ssoUrl: string, accessToken: SsoAccessToken): void
    invalidateAccessToken(ssoUrl: string): void
}

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoCache.kt
// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package software.aws.toolkits.jetbrains.core.credentials.sso

interface SsoCache {
    fun invalidateClientRegistration(ssoRegion: String)
    fun invalidateAccessToken(ssoUrl: String)

    fun loadClientRegistration(cacheKey: ClientRegistrationCacheKey): ClientRegistration?
    fun saveClientRegistration(cacheKey: ClientRegistrationCacheKey, registration: ClientRegistration)
    fun invalidateClientRegistration(cacheKey: ClientRegistrationCacheKey)

    fun loadAccessToken(cacheKey: AccessTokenCacheKey): AccessToken?
    fun saveAccessToken(cacheKey: AccessTokenCacheKey, accessToken: AccessToken)
    fun invalidateAccessToken(cacheKey: AccessTokenCacheKey)
}
